### PR TITLE
Infrastructure for tag-based publishing

### PR DESCRIPTION
**Pull request by lrytz**
Fixes https://github.com/scala/scala-collection-compat/issues/16

### CODE REVIEW

**Comment by lrytz** [comment]
@SethTisue I moved away from using the rather brittle `scalaVersionsByJvm` feature of the sbt module plugin. Instead, the travis build matrix sets the combinations of scala and jvm versions to test, and for those combinations that should be used in a release, it needs to make sure the `RELEASE_COMBO` var is `true`. Let me know what you think. Maybe it can be further simplified. I'd like to do something like this for the other modules as well.

**Comment by lrytz** [comment]
Tested releasing by creating a dummy tag (https://github.com/scala/scala-collection-compat/releases/tag/v0.0.1-TEST). It created 4 staging repos (it's still misterious to me why / how releases are split into multiple staging repos...)
  - https://oss.sonatype.org/content/repositories/orgscala-lang-1544/org/scala-lang/modules/
  - https://oss.sonatype.org/content/repositories/orgscala-lang-1545/org/scala-lang/modules/
  - https://oss.sonatype.org/content/repositories/orgscala-lang-1546/org/scala-lang/modules/
  - https://oss.sonatype.org/content/repositories/orgscala-lang-1547/org/scala-lang/modules/

So I think this is good to go.

File: project/plugins.sbt [hunk 1]
```diff
@@ -3,3 +3,4 @@ val scalaJSVersion =
 
 addSbtPlugin("org.scala-js" % "sbt-scalajs" % scalaJSVersion)
 addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "0.4.0")
+addSbtPlugin("org.scala-lang.modules" % "sbt-scala-module" % "1.0.12")
```
Comment (SethTisue) [inline project/plugins.sbt]: should be 1.0.14?

Comment (lrytz) [reply project/plugins.sbt]: Sure!

**Pull request merged**